explorer/explorertypes.go

```diff
@@ -414,6 +415,13 @@ func ReduceAddressHistory(addrHist []*dbtypes.AddressRow) *AddressInfo {
 			IsFunding: addrOut.IsFunding,
 		}
 
+		if tx.TxType == "" {
+			msgTx, err := txhelpers.MsgTxFromHex(addrOut.TxHash)
```

Comment:
The transaction hash (`TxHash`) is not the hex-encoded bytes of the entire transaction that is required by `MsgTxFromHex`.  However, pulling the entire transaction hex is not an efficient approach for this.  `AddressRow` has a `TxType` field, which should be set elsewhere, probably in `(*ChainDB).AddressHistory` -> `(*ChainDB).AddressTransactions`.  Note that each addresses table query called from there (e.g. `SelectAddressLimitNByAddress`) could just retrieve Tx Type because the addresses table has a `tx_type` column.  It merely needs to be retrieved along with the rest of the data by adding it to `addrsColumnNames` and modifying *all* functions using affected queries.

Reply:
I see, Thanks